source/_UIAHandler.py

```diff
@@ -224,9 +230,40 @@ def MTAThreadFunc(self):
 			self.MTAThreadInitException=e
 		finally:
 			self.MTAThreadInitEvent.set()
-		self.MTAThreadStopEvent.wait()
+		while True:
+			func=self.MTAThreadQueue.get()
+			if func:
+				func()
+			else:
+				break
 		self.clientObject.RemoveAllEventHandlers()
 
+	def _onForegroundChange(self):
+		pendingForegroundUIAElement=self.pendingForegroundUIAElement
+		if pendingForegroundUIAElement==self.currentForegroundUIAElement:
+			return
+		if self.currentForegroundUIAElement:
+			try:
+				self.clientObject.removePropertyChangedEventHandler(self.currentForegroundUIAElement,self)
+			except COMError:
+				# The old UIAElement died as the window was closed.
+				# The system should forget the old event registration itself.
+				pass
+		self.currentForegroundUIAElement=pendingForegroundUIAElement

```

Comment:
It might make sense to set pendingForegroundUIAElement to None here. My concern with the current implementation is that if onForegroundChange fails multiple times in a row to set the pendingForegroundUIAElement, the pendingForegroundUIAElement still refers to an old element for which unregistration is attempted multiple times as well.

Reply:
Err... I would argue rather that curForegroundUIAElement should be set to None if registration fails, not pendingForegroundUIAElement. 
PendingForegroundUIAElement is always replaced with a new element before _foregroundChange is queued.  CurForegroundUIAElement if it is left set will try to be unregistered multiple times.

Reply:
Yes, you're actually correct here. Sorry for that, but it at least triggered you to do the right thing here :)